app/controllers/posts_controller.rb

```diff
@@ -36,17 +35,34 @@ def update
 
   def destroy
     @post.destroy
-    redirect_to user_posts_path(@user), notice: 'Post was successfully destroyed.'
+    redirect_to user_posts_path(current_user), notice: 'Post was successfully destroyed.'
+  end
+
+  def like
+    if @post.increment!(:likes_counter)
+      redirect_to @post, notice: 'You liked the post!'
+    else
+      redirect_to root_path, alert: 'Post not found.'
+    end
+  end
```

Comment:
- [ ] Great work so far! 👏 I see that you're handling post likes within the post controller. While it's functional, I'd like to suggest a more organized approach.

    - Consider creating a dedicated `LikesController` to handle actions related to post likes. This approach adheres to the Single Responsibility Principle and makes your code more maintainable. By having a `LikesController`, you can centralize all operations related to likes, including the creation of likes for posts.

**Here's a quick suggestion:**

This route configuration allows you to handle post likes through the `LikesController`:

```ruby
# controllers/likes_controller.rb
class LikesController < ApplicationController
  def create
    # Logic to create a like for a post
  end
end
```

And your route would be something like:

```
POST /posts/:post_id/likes(.:format) => likes#create
```

This separation of concerns improves code organization and makes your application more scalable.

Happy coding! 😊🚀

Reply:
![imagen](https://github.com/jlberbesi/blog-app/assets/88105477/86a813bc-4a9d-4733-b609-71771e236db8)

I created the likes controller to have the application more scalable